storage/innobase/dict/dict0stats.cc

```diff
@@ -1372,6 +1406,91 @@ static dberr_t btr_pcur_open_level(btr_pcur_t *pcur, ulint level, mtr_t *mtr,
 }
 
 
+/** Statistics collected for a single B-tree level
+during index analysis. Used to track distinct key values,
+record counts, and page information when analyzing index
+levels for the query optimizer's statistics. */
+struct IndexLevelStats
+{
+  /** Level number */
+  ulint level;
+  /** Whether nulls are considered unequal for statistics */
+  bool nulls_unequal;
+  /** Array for number of distinct keys */
+  uint64_t* n_diff;
+  /** Array of number of non-null keys for all prefixes */
+  uint64_t* n_non_null;
+  /** Boundaries of groups of distinct keys */
+  boundaries_t* bounds;
```

Comment:
Formatting: space before `*`, not after it. (Please check the entire patch for this; I am not flagging every occurrence.)